Current bfish 2018 Oct 10

jingmeihu opened the pull request.

alexpatel commented:
Also, I've been taught that to distinguish between Pull Requests that are ready/not ready one should not assign anyone to the PR until it is ready to be reviewed/merged.

alexpatel commented:
Okay, so I have verified that caps.hl in the upstream code matches (roughly, because there are several more hierarchy layers scattered about) the dependency diagram from page 9 of the [capabilities management docs](http://www.barrelfish.org/publications/TN-013-CapabilityManagement.pdf):

![screen shot 2017-06-07 at 4 06 49 pm](https://user-images.githubusercontent.com/4894568/26899038-e0c48e92-4b9b-11e7-8876-d174de0572e0.png)

Next up is to hack together a Python script to grep through caps.hl and output a .dot file.

@mwookawa I am wondering why the code generated by hamlet/FoF (build/x86_64/capabilities/*.c) includes switch cases for all possible architectures - it seems like a lot of wasted instructions that it doesn't just output x86_64 case statements.

alexpatel commented:
@ming the PNG graph this script generates is [here](https://github.com/Harvard-PRINCESS/barrelfish/blob/9b0e8325cf8f4d4753b5df8429f5b8224a8ef344/capabilities/caps.png) (the aspect ratio of the image is off because the tree is so wide, you'll have to download and zoom in).

script is ready for review - there is some pretty-printing code in HamletBackend.hls that is also amenable to this task + a Haskell graphviz lib, if we want to upstream something

alexpatel commented:
Also, having two repositories means that none of the issues from Guppy will link over...

mwookawa commented:
@alexpatel all capabilities defined in the barrelfish codebase can be manipulated by a kernel running on any architecture whose monitor obeys the common interface. hence, hamlet generates capability types and retyping predicates for every architecture on every build. does that answer the switch/case verbosity question?

achreto committed:
k1om: the xeon phi does not support clflush instructions

Signed-off-by: Reto Achermann <reto.achermann@inf.ethz.ch>

achreto committed:
hake: adapt to the GHC 8.0 changes in interface

Signed-off-by: Reto Achermann <reto.achermann@inf.ethz.ch>

achreto committed:
hake: adapt to the GHC 8.0 changes in interface

Signed-off-by: Reto Achermann <reto.achermann@inf.ethz.ch>

adam-turowski committed:
fpu: unifing the fpu's context switching
x86_64: fixing the stack's alignment

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
fpu: unifing the fpu's context switching
x86_64: fixing the stack's alignment

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
thc: fixing the stack's alignment

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
thc: fixing the stack's alignment

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
fpu: some cleanups

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
fpu: some cleanups

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
Removing the unnecessary syscall GET_ABS_TIME from x86_64

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
Removing the unnecessary syscall GET_ABS_TIME from x86_64

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
pandaboard: fixing small compilation issues

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
pandaboard: fixing small compilation issues

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
pci: temporary workaround for over-4G-devices

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
pci: temporary workaround for over-4G-devices

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

simu committed:
harness: Update ETH pandaboard rack machine definitions to work on all emmentaler build hosts

Signed-off-by: Simon Gerber <simon.gerber@inf.ethz.ch>

achreto committed:
k1om: remove fpu_save and fpu_resore functions as they are in handled else where now

Signed-off-by: Reto Achermann <reto.achermann@inf.ethz.ch>

simu committed:
serial,pc16550: properly disable interrupts before polling on interrupt

Signed-off-by: Simon Gerber <simon.gerber@inf.ethz.ch>

haeckir91 committed:
Merge branch 'master' into '18.04'

Signed-off-by: Roni Häcki <roni.haecki@inf.ethz.ch>

haeckir91 committed:
harness: proc_mgmt test change cores used to depend on command line
input and not hard coded values

Signed-off-by: Roni Häcki <roni.haecki@inf.ethz.ch>

achreto committed:
acpi: provide implementation to dump the ACPI tables

Signed-off-by: Reto Achermann <reto.achermann@inf.ethz.ch>

Commit:
irqtest: Fix test. Enable multi instance

Enable to instantiate multiple irqtest in the same domain.

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

Commit:
ACPI: Enable debug printer (with preprocessor switch)

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

achreto committed:
libacpica: adding haek target for the dump facilities of the acpi tables

Signed-off-by: Reto Achermann <reto.achermann@inf.ethz.ch>

Commit:
harness: rename grubboot to boottest

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

Commit:
Merge branch 'master' into 18.04

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

Commit:
e1000: Fix for recent QEMU versions

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

Commit:
acpi: Fix build for ARMv7

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

adam-turowski committed:
arm: adding gic v3

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
armv8: adding support for NEON

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

haeckir91 committed:
kernel: added missing device dependency

Signed-off-by: Roni Häcki <roni.haecki@inf.ethz.ch>

haeckir91 committed:
kernel: do not initialze xmm registers on xeon phi

Xeon phi does not have xmm register leading to compilation errors

Signed-off-by: Roni Häcki <roni.haecki@inf.ethz.ch>

adam-turowski committed:
fputest: new, better test

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

Commit:
harness: Make compiler-rt tests run on qemu

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

Commit:
compiler-rt-tests: Retab file

GCC6 now warns on wrongly idented files

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

Commit:
save FPU state on disp_save

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

adam-turowski committed:
xeon_phi: fixing timers, dispatching and stack's alignment

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

haeckir91 committed:
gem5: fixed test for 18.04

Signed-off-by: Roni Häcki <roni.haecki@inf.ethz.ch>

haeckir91 committed:
gem5: wait time until we connect using telnet increased

Signed-off-by: Roni Häcki <roni.haecki@inf.ethz.ch>

adam-turowski committed:
debug: reverting debug_printf and acpi commits

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

Commit:
Use bios settings for bar configuration on QEMU

Recent QEMU versions, such as 2.11.1 shipped with Ubuntu 18.04, use BAR
addreses above the 32 bit boundary, hence we need to use the BIOS
provided PCI configuration.

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

Commit:
Make harness a bit more tolerant for accepting output

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

adam-turowski committed:
qemu: reverting few changes

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

Commit:
Make armv8 qemu wrapper work together with harness

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

Commit:
Restore interrupts on x86

Signed-off-by: Lukas Humbel <lukas.humbel@inf.ethz.ch>

adam-turowski committed:
arm: removing a debug print and fixing ACPI

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
armv7: fixing VFP support

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
armv7: fixing the a9's timer

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
armv7: fixing QEMU Zynq, adding 4-core QEMU armv7

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
arm: platform info cleanups, adding e1k to QEMU

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
armv7: fixing thread context switching

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
Merge branch '18.04'

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>

adam-turowski committed:
harness: changing the DS-5 directory

Signed-off-by: Adam Turowski <adam.turowski@inf.ethz.ch>
